vector-im/element-android
# [Live location sharing] Improve aggregation process of events

## Type of change


## Content



Creation of a dedicated Entity class named `LiveLocationAggregatedSummaryEntity` to aggregate and persist in local database the infos related to a live location share.
Bonus: fixing an issue with time units of the timestamps of the location events in `LocalEchoEventFactory`: timestamps should be in milliseconds.

## Motivation and context


Closes #5862 

During a live location share, we first receive a state event to indicate the live has starting for a certain duration. Then, we receive periodically location events. When user decides to stop the live, a new state event with the same state key is sent with a "live" field set to false.
We need to aggregate all these events into a simpler structure to ease the work on app side.

## Screenshots / GIFs





## Tests



- Update the app from a previous version
- Check there is no crash during the database migration

## Tested devices

- OS version(s): Android 11

## Checklist
